app/diagnostics.go

```diff
@@ -689,15 +683,19 @@ func (a *App) trackPlugins() {
 
 	pluginStates := a.Config().PluginSettings.PluginStates
 	plugins, _ := pluginsEnvironment.Available()
+	pluginTrackData := map[string]interface{}{}
```

Comment:
This should be a well defined struct instead of `interface{}`. Maybe something like just call the struct type `pluginTrackData`

Reply:
I was not sure if it merited a struct definition. Where do you suggest I define it (go newbie question here 🙋‍♀️)?

Reply:
In general you should avoid `interface{}` as it doesn't anything about the data and consumers have to guess what the type is.

In general all structs are placed in `model` so `app` and `store` can access it. But given that this struct will only be used in `app` it's fine to place it there.

Reply:
Agreed. My reasoning was that because it's going directly into sending the diagnostics to segment it was not intended to be consumed here by anyone

Reply:
While I see you point, I still think it's bad practice as it does make the code less readable.